### DISCUSSION TITLE
Using ${project.version} in artifact paths not possible?

### DISCUSSION THREAD

**Discussion by marceloverdijk** [jreleaser/jreleaser#639] [Q&A] [answered]
I'm using the distribution plugin to create distributions and jreleaser to publish releases to GitHub.

```
distributions {
    foo {
        distributionBaseName = "foo"
        contents {
            from "${buildDir}/data/foo.txt"
        }
    }
```

with the Gradle project version set to e.g. `1.2.0` in gradle.properties this will lead to the file: `build/distributions/foo-1.2.0.zip`

then I thought having:

```
jreleaser {
    files {
        artifact {
            path = "${buildDir}/distributions/foo-${project.version}.zip"
        }
    }
}
```

should be able to release it.

However it fails... as the GitHub action doing the release says:

```
Execution failed for task ':jreleaserFullRelease'.
> Path does not exist. build/distributions/foo-property 'version'.zip
```

when I hardcode the version in the path to:

```
jreleaser {
    files {
        artifact {
            path = "${buildDir}/distributions/foo-1.2.0.zip"
        }
    }
}
```

it does work.

Is there another smart way to make this "dynamic" as I have multiple artifact files and I want to avoid to change their name for every release...

**Comment by aalmiray** [comment, marked as answer]
That is because the `project` reference you see is actually of type `org.jreleaser.gradle.plugin.dsl.Project` (belonging to the `jreleaser` extension) which exposes a `version` property using Gradle's Property API. You have several choices:

1. store the value of Gradle's `project.version` in a dynamic property, then use it as `"${buildDir}/distributions/foo-$myDynVersion.zip"` for example.
2. invoke `project.version.get()` to retrieve the version of the `jreleaser.project` instance.
3. use the `{{projectVersion}}` mustache template.

Personally I prefer option 3, thus your code would look like

```groovy
jreleaser {
    files {
        artifact {
            path = "${buildDir}/distributions/foo-{{projectVersion}}.zip"
        }
    }
}
```

**Reply by marceloverdijk** [reply]
Thanks Andres, 
I wasn't aware of the project reference. Option 3 worked for me 👍 
Cheers,
Marcel

**Reply by aalmiray** [reply]
When in doubt prefer the use of mustache templates as their values are resolved much later than Gradle properties. May I suggest having a look at the list of available named templates https://jreleaser.org/guide/latest/configuration/name-templates.html? 😄
